File: app/src/main/java/khorunzhyicom/alex/theposter/presentation/activity/adapter/PosterTabsAdapter.java
package khorunzhyicom.alex.theposter.presentation.activity.adapter;


import android.support.annotation.NonNull;

import com.bluelinelabs.conductor.Controller;
import com.bluelinelabs.conductor.Router;
import com.bluelinelabs.conductor.RouterTransaction;
import com.bluelinelabs.conductor.support.RouterPagerAdapter;

import java.util.List;

public class PosterTabsAdapter extends RouterPagerAdapter {

    private final List<Controller> tabs;

    public PosterTabsAdapter(@NonNull Controller host, List<Controller> tabs) {
        super(host);
        this.tabs = tabs;
    }

    @Override
    public void configureRouter(@NonNull Router router, int position) {
        if (!router.hasRootController()) {
            Controller controller = tabs.get(position);
            router.pushController(RouterTransaction.with(controller));
        }
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return "movies";
    }

}

File: app/src/main/java/khorunzhyicom/alex/theposter/presentation/activity/adapter/PosterTabsAdapter.kt
package khorunzhyicom.alex.theposter.presentation.activity.adapter


import android.content.Context

import com.bluelinelabs.conductor.Controller
import com.bluelinelabs.conductor.Router
import com.bluelinelabs.conductor.RouterTransaction
import com.bluelinelabs.conductor.support.RouterPagerAdapter

class PosterTabsAdapter(host: Controller, val context: Context, val tabs: List<TabView>) : RouterPagerAdapter(host) {

    override fun configureRouter(router: Router, position: Int) {
        if (!router.hasRootController()) {
            val controller = tabs[position].tabController()
            router.pushController(RouterTransaction.with(controller))
        }
    }

    override fun getCount(): Int = tabs.size

    override fun getPageTitle(position: Int): CharSequence = context.getString(tabs[position].tabTitle())

}

